src/main.cc

```diff
@@ -112,23 +112,32 @@ void Application::split_path(const string& path, string& directory, string& file
     }
 }
 
+
+std::map<string, string> Application::get_rev_num_data() {
```

Comment:
Why using map where we can use a struct?
Declare appropriate struct in type_output.hh.